asterius/src/Asterius/CodeGen.hs

```diff
@@ -1139,74 +1139,192 @@ marshalCmmPrimCall (GHC.MO_U_QuotRem2 GHC.W64) [q, r] [lhsHi, lhsLo, rhs] = do
-                    mask32 rhsr 1,
-                    mask32 rhsr 0,
-                    ConstI32 1
-                  ],
-                callImportReturnTypes = [I32]
-              }
-            CallImport
-              { target' = "__asterius_quotrem2_quotient",
-                operands =
-                  [ mask32 lhsHir 1,
-                    mask32 lhsHir 0,
-                    mask32 lhsLor 1,
-                    mask32 lhsLor 0,
-                    mask32 rhsr 1,
-                    mask32 rhsr 0,
-                    ConstI32 0
-                  ],
-                callImportReturnTypes = [I32]
+  let quotout =
+        UnresolvedSetLocal
+          { unresolvedLocalReg = quotr,
+            value =
+              smash32IntTo64
+                CallImport
+                  { target' = "__asterius_quotrem2_quotient",
+                    operands =
+                      [ mask32 lhsHir 1,
+                        mask32 lhsHir 0,
+                        mask32 lhsLor 1,
+                        mask32 lhsLor 0,
+                        mask32 rhsr 1,
+                        mask32 rhsr 0,
+                        ConstI32 1
+                      ],
+                    callImportReturnTypes = [I32]
+                  }
+                CallImport
+                  { target' = "__asterius_quotrem2_quotient",
+                    operands =
+                      [ mask32 lhsHir 1,
+                        mask32 lhsHir 0,
+                        mask32 lhsLor 1,
+                        mask32 lhsLor 0,
+                        mask32 rhsr 1,
+                        mask32 rhsr 0,
+                        ConstI32 0
+                      ],
+                    callImportReturnTypes = [I32]
+                  }
+          }
+  let remout =
+        UnresolvedSetLocal
+          { unresolvedLocalReg = remr,
+            value =
+              smash32IntTo64
+                CallImport
+                  { target' = "__asterius_quotrem2_remainder",
+                    operands =
+                      [ mask32 lhsHir 1,
+                        mask32 lhsHir 0,
+                        mask32 lhsLor 1,
+                        mask32 lhsLor 0,
+                        mask32 rhsr 1,
+                        mask32 rhsr 0,
+                        ConstI32 1
+                      ],
+                    callImportReturnTypes = [I32]
+                  }
+                CallImport
+                  { target' = "__asterius_quotrem2_remainder",
+                    operands =
+                      [ mask32 lhsHir 1,
+                        mask32 lhsHir 0,
+                        mask32 lhsLor 1,
+                        mask32 lhsLor 0,
+                        mask32 rhsr 1,
+                        mask32 rhsr 0,
+                        ConstI32 0
+                      ],
+                    callImportReturnTypes = [I32]
+                  }
+          }
+  pure [quotout, remout]
+-- Atomic operations
```

Comment:
The diffs before this line seem to be formatting noise. Can you reset the parts above?

Reply:
Yeah, I overlooked the grouping by `git add --patch`, it should be fixed now.